istio-proxy/fetch.sh

```diff
@@ -272,7 +272,7 @@ function replace_ssl() {
 
       git clone http://github.com/maistra/envoy-openssl -b ${PROXY_GIT_BRANCH}
       pushd envoy-openssl
-        ./openssl.sh ${CACHE_DIR}/base/external/envoy OPENSSL
+        ./openssl.sh ${CACHE_DIR}/base/external/envoy OPENSSL ${SHA}
```

Comment:
If I understand correctly, this $SHA variable is only set within fetch() only if the proxy repo is cloned. Is it safe to reference it here? Better to make sure $SHA is always defined?

Reply:
I'll clean it up so that it's always cloned. The code that's checking to see if there's an existing proxy is old and shouldn't really be there. Thanks